node-notes-cli/app.ts

```diff
@@ -0,0 +1,54 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const [, , note] = process.argv;
+
+// creating the object type
+
+type Data = {
+  nextId: number;
+  notes: Record<string, string>;
+};
+
+// creating the statments for the try and catch statments
+
+async function writeFiles(data: Data): Promise<void> {
+  data.nextId += 1;
+  await writeFile('data.json', JSON.stringify(data), { encoding: 'utf8' });
+}
+
+async function createNodes(note: string): Promise<Data> {
```

Comment:
The names should probably be "notes" instead of "nodes". Also, it only creates one note, so the function name should be singular.